imports/api/pages/methods.js

```diff
@@ -68,3 +80,11 @@ function savePage (page, project) {
   Projects.update(project._id, { $set: githubProjectUpdate })
   return Pages.update(page._id, { $set: githubPageUpdate })
 }
+
+function uploadcareUuidFromUrl (uploadcareUrl) {
+  const invalidError = new Error(`Invalid Uploadcare URL: ${uploadcareUrl}`)
+  if (!uploadcareUrl) throw invalidError
+  const thisUrl = url.parse(uploadcareUrl)
+  if (thisUrl.hostname !== 'ucarecdn.com' || !thisUrl.pathname) throw invalidError
```

Comment:
This same logic caused a bug in study bundles when uploadcare changed their url structure. The solution was a regex that picks a valid uuid out of any string rather, which is similar to the spirit of what we doing here.
https://github.com/tableflip/study-bundles/blob/96a84ba6498ac4facd9a9982af9467c7ec00aaf5/packages/uploadcare/uploadcare.js#L18